src/rvsdg/optimize_direct_jumps.rs

```diff
@@ -0,0 +1,113 @@
+//! This module takes as input a [`SimpleCfgProgram`] and gets rid of any
+//! direct jumps between blocks.
+//! The block that is jumped to must have only one predecessor.
+//! This is used by `to_cfg` to clean up
+//! the output.
+
+use hashbrown::HashMap;
+use petgraph::{
+    stable_graph::{NodeIndex, StableDiGraph, StableGraph},
+    visit::Bfs,
+    visit::EdgeRef,
+    Direction,
+};
+
+use crate::cfg::{BasicBlock, Branch, Simple, SimpleCfgFunction, SimpleCfgProgram};
+
+struct JumpOptimizer<'a> {
+    simple_func: &'a SimpleCfgFunction,
+}
+
+impl<'a> JumpOptimizer<'a> {
+    fn optimized(&mut self) -> SimpleCfgFunction {
```

Comment:
Why have a separate struct / method here? Since there is only a single method for JumpOptimizer, would it be easier for this to be a standard function instead?

Reply:
One possibility I can imagine is that "removing redundant copies" could get added to this same struct / maybe they could share some additional state. But that could just as easily be its own file.

Reply:
So I've found myself spending a weird amount of time refactoring code to follow this pattern, and I think it's easier to set it up like this in the first place. For example, I did this to my RVSDG->CFG code, and a bunch of yihong's code.